### PR TITLE
Front End styling tidied up for Search Bar and Nav Links in the Header

### PR DESCRIPTION
**Pull request by samuelgoddard**
I noticed the nav in the header had a few small issues at narrow sizes (740px etc) where the Nav Links are overlapping the logo, and focusing the Search Bar breaks out of the layout.

I've reduced the size of the padding on the Nav Links and changed the styling of the focused Search Bar to absolutely position over the top of the Nav at Narrow breakpoints.

Let me know what you think. Cheers

### CODE REVIEW

**Comment by yyx990803** [comment]
Can you add a before/after screenshot for easier review?

**Comment by samuelgoddard** [comment]
Sure

Before (unfocused search)
before-unfocused
After (unfocused search)

after-unfocused

Before (focused search)
before-focused

After (focused search)
after-focused

When I get half an hour I'll put in a more solid fix as there's some other funky stuff going on with the nav when there's a tonne of nav items, it would be better to drop them onto another line that was it's infinitely scalable

Cheers

File: lib/default-theme/SearchBox.vue [hunk 1]
```diff
@@ -184,17 +184,40 @@ export default {
...
       border-color transparent
-      position relative
-      left 1rem
+      background-color white
+      z-index 10
+      &:focus
+        cursor text
+        width 10rem
+
+@media (min-width: $MQMobile) and (max-width: $MQNarrow)
```
Comment (yyx990803) [inline lib/default-theme/SearchBox.vue]: Currently there are multiple media queries with duplicated rules that can be merged together. e.g. This is the exact same media query as the one on line 223.

File: lib/default-theme/SearchBox.vue [hunk 2]
```diff
@@ -184,17 +184,40 @@ export default {
       a
         color $accentColor
 
-@media (max-width: $MQNarrow)
+@media (max-width: $MQMobile)
```
Comment (yyx990803) [inline lib/default-theme/SearchBox.vue]: This should be merged with the media query on line 228

File: lib/default-theme/SearchBox.vue [hunk 3]
```diff
@@ -184,17 +184,40 @@ export default {
...
+    margin-right 0
+    input
+      cursor pointer
+      width 0
+      border-color transparent
+      position absolute
+      background-color white
+      z-index 10
+      left 0
+      top: 0
```
Comment (meteorlxy) [inline lib/default-theme/SearchBox.vue]: revemove the `:`

**Comment by samuelgoddard** [comment]
@yyx990803 - changes pushed and ready for review

**Comment by chrisvfritz** [comment]
@michalsnik @shentao This is affecting a lot of Vue docs. Would either of you be able to help take over this PR and push it to completion? 🙂

**Comment by ulivz** [comment]
@chrisvfritz Sorry for the delay, I have fixed this issue but I need more condition-coverage test, will try to release it at ASAP.

**Comment by chrisvfritz** [comment]
@ulivz Thank you! I appreciate your work. 🙂

**Comment by ulivz** [comment]
Closing it since we had a better fix at #714.

**Pull request closed without merging**